src/test/java/com/purbon/kafka/topology/integration/SaslPlaintextKafkaContainer.java

```diff
@@ -0,0 +1,142 @@
+    withEnv("KAFKA_OPTS", "-Djava.security.auth.login.config=/tmp/broker_jaas.conf");
+  }
+
+  public String getBootstrapServers() {
+    if (port == PORT_NOT_ASSIGNED) {
+      throw new IllegalStateException("You should start Kafka container first");
+    }
+    return String.format("%s:%s", getHost(), port);
+  }
+
+  @Override
+  protected void doStart() {
+    withCommand(
+        "sh", "-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
+    super.doStart();
+  }
+
+  @Override
+  protected void containerIsStarting(
+      final InspectContainerResponse containerInfo, final boolean reused) {
+    super.containerIsStarting(containerInfo, reused);
+    port = getMappedPort(KAFKA_PORT);
+    if (reused) {
+      return;
+    }
+    final String zookeeperConnect = startZookeeper();
+    final String advertisedListeners =
+        LISTENERS
+            .replaceAll(":" + KAFKA_PORT, ":" + port)
+            .replaceAll("0\\.0\\.0\\.0", getContainerIpAddress());
+    final String command =
+        "#!/bin/bash\n"
+            + "cat > /tmp/broker_jaas.conf <<EOT\n"
+            + "KafkaServer { org.apache.kafka.common.security.plain.PlainLoginModule required username=\"kafka\" password=\"kafka\" user_kafka=\"kafka\" user_alice=\"alice-secret\" user_bob=\"bob-secret\"; };\n"
+            + "EOT\n"
+            + "export KAFKA_ZOOKEEPER_CONNECT='"
+            + zookeeperConnect
+            + "'\n"
+            + "export KAFKA_ADVERTISED_LISTENERS='"
+            + advertisedListeners
+            + "'\n"
+            + ". /etc/confluent/docker/bash-config\n"
+            + "/etc/confluent/docker/configure\n"
+            + "/etc/confluent/docker/launch\n";
+    copyFileToContainer(
+        Transferable.of(command.getBytes(StandardCharsets.UTF_8), 0755), STARTER_SCRIPT);
+  }
+
+  private String startZookeeper() {
+    final ExecCreateCmdResponse execCreateCmdResponse =
+        dockerClient
+            .execCreateCmd(getContainerId())
+            .withCmd(
+                "sh",
+                "-c",
+                "echo '*** Starting Zookeeper'\n"
+                    + "printf 'clientPort="
+                    + ZOOKEEPER_PORT
+                    + "\n"
+                    + "dataDir=/var/lib/zookeeper/data\ndataLogDir=/var/lib/zookeeper/log' > zookeeper.properties\n"
+                    + "zookeeper-server-start zookeeper.properties\n")
+            .withAttachStderr(true)
+            .withAttachStdout(true)
+            .exec();
+    try {
+      dockerClient
+          .execStartCmd(execCreateCmdResponse.getId())
+          .start()
+          .awaitStarted(10, TimeUnit.SECONDS);
+    } catch (final InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+    return "127.0.0.1:" + ZOOKEEPER_PORT;
+  }
+
+  public Admin getAdmin() {
```

Comment:
```suggestion
  public AdminClient getAdminClient() {
```

Reply:
AdminClient has a class-level comment stating
```
 * This class may be removed in a later release, but has not be marked as deprecated to avoid unnecessary noise.
```
I will make the change, but it may have to be changed again in the future.